app/api/user/getuser_by_wallet/route.ts

```diff
@@ -0,0 +1,206 @@
+import { MIDDLEWARE_CONFIG, LIVEPEER_CONFIG } from "@/services/config.service";
+import { NextResponse } from "next/server";
+import { conn } from "@/services/mongo.service";
+import { addUser, findOne, findOneAndUpdate } from "@/utils/user/user";
+import { Str } from "@supercharge/strings/dist";
+import axios from "axios";
+import jwt from "jsonwebtoken";
+
+export async function POST(request: Request) {
+  try {
+    await conn();
+    const req = await request.json();
+    const wallet_id = req.walletId;
+    const name = req.name;
+    const email = req.email;
+    const profileImage = req.profileImage;
+
+    const referrer = req.referrer;
+
+    const userData = await findOne({ wallet_id: wallet_id });
+    if (userData) {
+      const data = {
+        user_id: userData._id,
+      };
+      const authtoken = jwt.sign(data, MIDDLEWARE_CONFIG.JWT_SECRET);
+      let loginData = {
+        user: userData,
+        jwtToken: authtoken,
+      };
+      return NextResponse.json({
+        status: "success",
+        data: loginData,
+      });
+    } else if (!userData && req.fetchData) {
+      return "No user found";
+    } else {
+      if (email != "" && email != null && email != undefined) {
+        let userName = email.substring(0, email.lastIndexOf("@"));
+        console.log(userName);
```

Comment:
Remove unnecessary console logs, or add some correct names to the log if necessary to check in production logs.